doc/book/admin/access_control.rst

```diff
@@ -1,377 +1,1069 @@
+
+..  code-block:: lua
+
+    box.schema.user.grant('testuser','execute','function','sum')
+
+
+
+
+
+.. _access_control_grant_users:
+
+Users
+~~~~~
+
+In this example, ``testuser`` gets privileges to create other users:
+
+..  code-block:: lua
+
+    box.schema.user.grant('testuser','create','user')
+    box.schema.user.grant('testuser', 'read,write', 'space', '_user')
+    box.schema.user.grant('testuser', 'write', 'space', '_priv')
+
+.. _access_control_grant_roles:
+
+Roles
+~~~~~
+
+To let ``testuser`` create new roles, grant the following privileges:
+
+..  code-block:: lua
+
+    box.schema.user.grant('testuser','create','role')
+    box.schema.user.grant('testuser', 'read,write', 'space', '_user')
+    box.schema.user.grant('testuser', 'write', 'space', '_priv')
+
+
+.. _access_control_grant_execute_code:
+
+Executing code
+~~~~~~~~~~~~~~
+
+To let ``testuser`` execute Lua code, grant the ``execute`` privilege to the ``lua_eval`` object:
+
+..  code-block:: lua
+
+    box.schema.user.grant('testuser','execute','lua_eval')
+
+Similarly, executing an arbitrary SQL expression requires the ``execute`` privilege to the ``sql`` object:
+
+..  code-block:: lua
+
+    box.schema.user.grant('testuser','execute','sql')
+
+
+
+
+
+.. _creating_users_and_objects_granting_privileges:
+
+Example
+~~~~~~~
+
+In the example below, the created Lua function is executed under the user ID of its
+creator, even if called by another user.
+
+First, the two spaces (``space1`` and ``space2``) are created, and a no-password user (``private_user``)
+is granted full access to them. Then ``read_and_modify`` is defined and the
+no-password user becomes this function's creator. Finally, another user
+(``public_user``) is granted access to execute Lua functions created by the no-password user.
+
+..  code-block:: lua
+
+    box.schema.space.create('space1')
+    box.schema.space.create('space2')
+    box.space.space1:create_index('pk')
+    box.space.space2:create_index('pk')
+
+    box.schema.user.create('private_user')
+
+    box.schema.user.grant('private_user', 'read,write', 'space', 'space1')
+    box.schema.user.grant('private_user', 'read,write', 'space', 'space2')
+    box.schema.user.grant('private_user', 'create', 'universe')
+    box.schema.user.grant('private_user', 'read,write', 'space', '_func')
+
+    function read_and_modify(key)
+      local space1 = box.space.space1
+      local space2 = box.space.space2
+      local fiber = require('fiber')
+      local t = space1:get{key}
+      if t ~= nil then
+        space1:put{key, box.session.uid()}
+        space2:put{key, fiber.time()}
+      end
+    end
+
+    box.session.su('private_user')
+    box.schema.func.create('read_and_modify', {setuid= true})
+    box.session.su('admin')
+    box.schema.user.create('public_user', {password = 'secret'})
+    box.schema.user.grant('public_user', 'execute', 'function', 'read_and_modify')
```

Comment:
Let's maybe write a conclusion sentence, smth like: `now whenever public_user calls the function, it is executed on behalf of its creator, private_user`